Update the minimum supported NumPy to 1.24 (#9739)
numba/numba: docs/upcoming_changes/9739.highlight.rst

> Updated the minimum supported NumPy to 1.24
> -------------------------------------------

Reviewer: ```suggestion
Update the minimum supported NumPy version to 1.24
--------------------------------------------------
```

> This release updates the minimum official supported version for NumPy to 1.24 and later.

Reviewer: ```suggestion
This release updates the minimum supported version of NumPy to 1.24.
```
